### PR TITLE
feat: maintenance_window support

### PR DESCRIPTION
**Pull request by czerasz-mineiros**

### CODE REVIEW

**Review by mariux — COMMENTED** [review]
thanks for the added feature.. please add
- the new variable to unit-complete test
- update the documentation new variable
- update the change log and prepare a release by updating version in readme.

thanks a lot..

**Comment by soerenmartius** [comment]
can we still finish this PR? :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +34,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Initial Implementation
 
 [unreleased]: https://github.com/mineiros-io/terraform-google-composer/compare/v0.1.0...HEAD
```
Comment (mariux) [inline CHANGELOG.md:36]:
```suggestion
[unreleased]: https://github.com/mineiros-io/terraform-google-composer/compare/v0.2.0...HEAD
```

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -407,6 +407,32 @@ section {
             Customer-managed Encryption Key available through Google's Key Management Service. It must be the fully qualified resource name, i.e. `projects/project-id/locations/location/keyRings/keyring/cryptoKeys/key`. Cannot be updated.
           END
         }
+
+        variable "maintenance_window" {
+          type        = object(maintenance_window)
+          description = "(Optional) The configuration settings for Cloud Composer maintenance windows."
```
Comment (mariux) [inline README.tfdoc.hcl:413]:
```suggestion
          description = "The configuration settings for Cloud Composer maintenance windows."
```

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -407,6 +407,32 @@ section {
...
+
+          attribute "start_time" {
+            type        = string
+            description = <<-END
+              (Required) Start time of the first recurrence of the maintenance window
```
Comment (mariux) [inline README.tfdoc.hcl:418]:
```suggestion
            type        = string
            required    = true
            description = <<-END
              Start time of the first recurrence of the maintenance window
```

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -407,6 +407,32 @@ section {
...
+            END
+          }
+
+          attribute "end_time" {
+            type        = string
+            description = <<-END
+              (Required) Maintenance window end time. It is used only to calculate the duration of the maintenance window. The value for end-time must be in the future, relative to 'start_time'.
```
Comment (mariux) [inline README.tfdoc.hcl:425]:
```suggestion
            type        = string
            required    = true
            description = <<-END
              Maintenance window end time. It is used only to calculate the duration of the maintenance window. The value for end-time must be in the future, relative to 'start_time'.
```

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -407,6 +407,32 @@ section {
...
+            END
+          }
+
+          attribute "recurrence" {
+            type        = string
+            description = <<-END
+              (Required) Maintenance window recurrence. Format is a subset of RFC-5545 (https://tools.ietf.org/html/rfc5545) 'RRULE'. The only allowed values for 'FREQ' field are 'FREQ=DAILY' and 'FREQ=WEEKLY;BYDAY=…'. Example values: 'FREQ=WEEKLY;BYDAY=TU,WE', 'FREQ=DAILY'.
```
Comment (mariux) [inline README.tfdoc.hcl:432]:
```suggestion
            type        = string
            required    = true
            description = <<-END
              Maintenance window recurrence. Format is a subset of [RFC-5545](https://tools.ietf.org/html/rfc5545) `RRULE`. The only allowed values for `FREQ` field are `FREQ=DAILY` and `FREQ=WEEKLY;BYDAY=…`. Example values: `FREQ=WEEKLY;BYDAY=TU,WE`, `FREQ=DAILY`.
```

File: main.tf [hunk 6]
```diff
@@ -110,5 +110,14 @@ resource "google_composer_environment" "composer_environment" {
         kms_key_name = var.kms_key_name # - (Required) Customer-managed Encryption Key available through Google's Key Management Service. It must be the fully qualified resource name, i.e. projects/project-id/locations/location/keyRings/keyring/cryptoKeys/key. Cannot be updated.
       }
     }
+
+    dynamic "maintenance_window" {
+      for_each = var.maintenance_window != null ? [1] : []
```
Comment (mariux) [inline main.tf:115]:
```suggestion
      for_each = var.maintenance_window != null ? [var.maintenance_window] : []
```

File: main.tf [hunk 7]
```diff
@@ -110,5 +110,14 @@ resource "google_composer_environment" "composer_environment" {
...
+      content {
+        start_time = try(var.maintenance_window.start_time, null)
+        end_time   = try(var.maintenance_window.end_time, null)
+        recurrence = try(var.maintenance_window.recurrence, null)
```
Comment (mariux) [inline main.tf:119]: if those are required, why do we default to null?

```suggestion
        start_time = try(maintenance_window.value.start_time, null)
        end_time   = try(maintenance_window.value.end_time, null)
        recurrence = try(maintenance_window.value.recurrence, null)
```

**Review by mariux — CHANGES_REQUESTED** [review]
some minor changes and a question about required variables..

**Review by mariux — CHANGES_REQUESTED** [review]
the requested changes are not fully implemented the variables in the the block are not marked required yet.

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**